### PR TITLE
Stop event-polling for an account when it's logged out

### PR DESCRIPTION
**Pull request by chrisbobbe**
And make a few other cleanups and comments.

Fixes: #5022
Related: #5009

### CODE REVIEW

**Comment by chrisbobbe** [comment]
I've just rebased this past #5259. I'd appreciate a review! 🙂

**Review by gnprice — COMMENTED** [review]
Thanks! Comments below.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed. 🙂

File: src/session/sessionReducer.js [hunk 1]
```diff
@@ -24,6 +24,16 @@ import { getHasAuth } from '../account/accountsSelectors';
  * See {@link SessionState} for discussion of what "non-persistent" means.
  */
 export type PerAccountSessionState = $ReadOnly<{
+  /**
+   * The event queue ID that we're currently polling on, if any.
```
Comment (gnprice) [inline src/session/sessionReducer.js:28]: I like this summary line! This seems like a good basis for thinking about what the behavior should be.

File: src/session/sessionReducer.js [hunk 2]
```diff
@@ -24,6 +24,16 @@ import { getHasAuth } from '../account/accountsSelectors';
...
+   *
+   * Null when we're not polling on any event queue:
+   * - Between startup and registering a queue
+   * - After the server tells us our old queue was invalid, and before we've
+   *   registered a new one
+   * - While this account is logged out
+   * - While this account is logged in but is not the active account
```
Comment (gnprice) [inline src/session/sessionReducer.js:35]: I don't think this item applies.

* At present, the only per-account state we have is the one for the active account. So "this account" here is always the active account.
* In a future where we have any per-account session state for accounts other than the active account, I think that will go together with maintaining an event queue for those accounts.

File: src/session/sessionReducer.js [hunk 3]
```diff
@@ -162,26 +172,43 @@ export default (state: SessionState = initialState, action: Action): SessionStat
...
       };
 
     case LOGOUT:
       return {
         ...state,
         needsInitialFetch: false,
         loading: false,
+
+        // Although it may be valid, forget the event queue ID -- we know
+        // the user won't care about it.
```
Comment (gnprice) [inline src/session/sessionReducer.js:198]: I'd frame this a bit differently:
```suggestion
        // Stop polling this event queue.
```

File: src/session/sessionReducer.js [hunk 4]
```diff
@@ -162,26 +172,43 @@ export default (state: SessionState = initialState, action: Action): SessionStat
...
 
     case ACCOUNT_SWITCH:
       return {
         ...state,
         needsInitialFetch: true,
         loading: false,
+
+        // Although it may be valid, forget the old account's event queue --
+        // we know the user won't care about it. We also know we're about to
+        // request a new event queue; no use hanging on to the old one.
```
Comment (gnprice) [inline src/session/sessionReducer.js:210]: Similarly:

```suggestion
        // Stop polling this event queue.  (We'll request a new one soon,
        // for the new account.)
```

**Review by gnprice — COMMENTED** [review]
Thanks for the revision! Just small comments below.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

File: src/session/sessionReducer.js (hunk 1 above)
Comment (gnprice) [reply src/session/sessionReducer.js:28]: Yeah. Given that in that situation when the request eventually finishes, we're going to throw away the result without doing anything with it, I think it's still fair to say that we're not "currently polling on" that queue -- we do still have an HTTP request open, but as far as the app's logical behavior is concerned it already may as well not exist.

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Pull request merged**